src/storage/gc_worker.rs

```diff
@@ -251,8 +259,12 @@ impl<E: Engine> GCRunner<E> {
         }
         self.stats.add_statistics(&txn.take_statistics());
 
+        let write_size = txn.write_size();
         let modifies = txn.into_modifies();
         if !modifies.is_empty() {
+            if let Some(limiter) = &self.limiter {
+                limiter.request(write_size as i64);
```

Comment:
The gc_worker will be blocked here if the limit exceeded, is it expected?

Reply:
yes

Reply:
I think yes. We may avoid the blocking when refactoring it with futures.